### PR TITLE
Port user test cases

### LINKED ISSUE #6399: WebUI test coverage: Users

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #6399
```
pytest tests/foreman/ui_airgun/test_user.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-03-07 11:57:11 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_user.py .                                                                                                                                                 [100%]

=========================================================================== 1 passed, 2 warnings in 200.35 seconds ===============================================
```
```
pytest tests/foreman/ui_airgun/test_ldap_authentication.py -k 'test_positive_create_user_in_ldap_mode or test_positive_login_ad_user_no_roles or test_positive_login_ad_user_basic_roles'
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-03-07 12:04:19 - conftest - DEBUG - BZ deselect is disabled in settings

collected 14 items / 11 deselected                                                                                                                                                           

tests/foreman/ui_airgun/test_ldap_authentication.py ...                                                                                                                                [100%]

========================================================================= 3 passed, 11 deselected in 276.30 seconds =========================================================================
```

### CODE REVIEW

File: tests/foreman/ui_airgun/test_ldap_authentication.py [hunk 1]
```diff
@@ -573,6 +577,7 @@ def test_positive_update_external_user_roles(
         assert current_user == ldap_data['ldap_user_name']
 
 
+@skip_if_not_set('ldap')
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_ldap_authentication.py]: please replace with
```python
if not setting_is_set('ldap'):
    skip('skipping tests due to missing ldap settings', allow_module_level=True)
```
at the beginning of this module

File: tests/foreman/ui_airgun/test_ldap_authentication.py [hunk 2]
```diff
@@ -763,3 +770,88 @@ def test_positive_add_katello_role_with_org(
...
+    auth_source_name = 'LDAP-' + auth_source.name
+    name = gen_string('alpha')
+    with session:
+        session.user.create({
+            'user.login': name,
+            'user.auth': auth_source_name,
+        })
+        assert session.user.search(name)[0]['Username'] == name
+        user_values = session.user.read(name)
+        assert user_values['user']['auth'] == auth_source_name
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_ldap_authentication.py]: shouldn't we perform some sort of cleanup in the end not to spam ldap server with pointless users?

File: tests/foreman/ui_airgun/test_ldap_authentication.py [hunk 3]
```diff
@@ -763,3 +770,88 @@ def test_positive_add_katello_role_with_org(
...
+
+    :CaseLevel: Integration
+    """
+    with Session(
+            test_name,
+            ldap_data['ldap_user_name'],
+            ldap_data['ldap_user_passwd'],
+    ) as ldapsession:
+        with raises(NavigationTriesExceeded):
+            ldapsession.user.search('')
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_ldap_authentication.py]: can we try to read current user somehow, to ensure expected result `Log in to foreman UI successfully`?

File: tests/foreman/ui_airgun/test_ldap_authentication.py (hunk 2 above)
Comment (oshtaier) [reply tests/foreman/ui_airgun/test_ldap_authentication.py]: it will create user only on local Satellite server, just use ldap authentication for its work

**Comment by oshtaier** [comment]
```
pytest tests/foreman/ui_airgun/test_ldap_authentication.py::test_positive_login_ad_user_no_roles
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.1, py-1.5.3, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.2.1, mock-1.6.3
collecting ... 2019-03-07 12:57:40 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui_airgun/test_ldap_authentication.py .                                                                                                                                  [100%]

================================================================================= 1 passed in 65.13 seconds ==============================================================================
```

**Pull request merged**